### ISSUE TITLE
DCO - Expected — Waiting for status to be reported

### ISSUE THREAD

**Issue by AOrthVector** [dcoapp/app#218]
One of my project's repositories first had this issue 27 days ago. It was 3 weeks before I noticed my changes were not merged. I did see issue 211 and thought my problem would be solved, but this is not the case.

This is the [first repository](https://github.com/genevaers/Community) to encounter the problem. This [PR](https://github.com/genevaers/Community/pull/319) details some of the different things I tried.

I was the only person running into this problem on the GenevaERS Community repository.

Last week, I learned another [project repository](https://github.com/genevaers/User-Documentation) is having the problem intermittently,

https://github.com/dcoapp/app/issues/211

**Comment by davidLif** [comment]
Did you ever find a solution?

**Comment by Loirooriol** [comment]
This happens to me very frequently in https://github.com/servo/servo/

What tends to work is using the Github interface to rebase the PR:

<img width="794" height="391" alt="Image" src="https://github.com/user-attachments/assets/b1dd6d63-9d56-46b6-9e7a-1e3e17b0a5c1" />

Rebasing locally and force-pushing doesn't seem to work.

It's annoying and wasteful, because rebasing implies running the other CI checks again.

**Comment by tykeal** [comment]
**A recovery mechanism for stuck checks has shipped 🎉**

This is the same symptom tracked in #162: if the app misses a webhook delivery (for example during an outage or restart), it never creates the DCO check run for that PR, so a *required* DCO check sits on **"Expected — Waiting for status to be reported"** with no button in the UI to re-request it.

There is now a self-service fix. **Commenting `@dcoapp recheck` on the affected (open) pull request** forces the app to (re-)create and run the DCO check, unblocking the PR — even when the original webhook was missed. See #288 (closes #283).

This is meant to replace the awkward workarounds mentioned in this thread: you no longer need to rebase from the GitHub UI (which re-runs *all* of your CI) just to nudge the DCO check. The comment re-runs only the DCO check.

- Anyone can trigger it; it only acts on open PRs and only on that exact comment.
- It requires **no new permissions** for existing installations — maintainers just need to ensure the **"Issue comment"** event is enabled on their installation (enabled by default going forward).

This lands alongside broader reliability work to reduce how often the app misses events in the first place (a Probot v14 upgrade and runtime fixes, plus post-deployment smoke tests). Outages can still happen, but they should now be both rarer and **recoverable** without maintainer intervention.

Closing as resolved by #288. If you hit a stuck check that `@dcoapp recheck` does **not** clear, please open a new issue with the PR link.

**Comment by Loirooriol** [comment]
Sadly, `@dcoapp recheck` doesn't work: #289

**Comment by tykeal** [comment]
**Correction to my earlier comment on the `@dcoapp recheck` command.**

The command shipped as a **pull request review** action, not a plain conversation comment (the comment-based version couldn't be enabled without an additional permission that would have forced every install to re-approve). On an open PR:

- Submit a **pull request review** with `@dcoapp recheck` on its own line in the review summary (Files changed → Review changes → Comment → Submit), **or**
- Add an **inline review comment** containing `@dcoapp recheck` on its own line.

Either re-runs (or first-creates) the DCO check to recover a PR stuck "waiting for status." See #289 for details. (Please open a new issue if you hit any problems.)
